File: Data Structures & Algorithms/house-robber/submission-4.java
class Solution {
    public int rob(int[] nums) {
        int rob1 = 0;
        int rob2 = 0;

        for (int num : nums) {
            int cur = Math.max(rob2, rob1 + num);
            rob1 = rob2;
            rob2 = cur;
        }

        return rob2;
    }
}

File: Data Structures & Algorithms/duplicate-integer/submission-4.cpp

class Solution {
public:
  bool hasDuplicate(vector<int>& nums) {
    set<int> duplicate;
    for (const int& num : nums) {
      if (duplicate.count(num)) {
        return true;
      }

      duplicate.insert(num);
    }
    return false;
  }
};